pkg/controller/managedcluster/managedcluster_controller.go

```diff
@@ -93,18 +93,29 @@ func (r *ReconcileManagedCluster) Reconcile(ctx context.Context, request reconci
 		return reconcile.Result{}, err
 	}
 
+	if ns.Labels == nil {
+		ns.Labels = map[string]string{}
+	}
 	modified := resourcemerge.BoolPtr(false)
 	// TODO: use one cluster label to filter the cluster ns.
 	// in ocm we use open-cluster-management.io/cluster-name label to filter cluster ns,
 	// but in acm we use cluster.open-cluster-management.io/managedCluster to filter cluster ns.
 	// to make sure the cluster ns can be filtered in some cases, add the 2 labels to the cluster ns here.
 	resourcemerge.MergeMap(modified, &ns.Labels, map[string]string{ClusterLabel: managedCluster.Name,
 		clusterv1.ClusterNameLabelKey: managedCluster.Name})
-
 	if !*modified {
 		return reconcile.Result{}, nil
 	}
 
+	if ns.Annotations == nil {
+		ns.Annotations = map[string]string{}
+	}
+	if _, ok := managedCluster.Annotations[constants.AnnotationRemainNamespace]; ok {
+		ns.Annotations[constants.AnnotationRemainNamespace] = ""
```

Comment:
Is there a specific reason why the annotation can be added to both the `ManagedCluster` and the cluster `Namespace`? In GitOps scenarios, if a user adds the annotation only to the Namespace and not the `ManagedCluster`, it could trigger an endless loop of annotation being added and removed on the Namespace.

Reply:
this feature requires the user to add the annotation to the cluster not the ns. 
import-controller will sync the annotation to the ns from the cluster because there is a case that the cluster is detached but there are other resources like clusterdeployment still in the ns, in this case need the annotation to determine if delete the ns after these resources are deleted.

Reply:
Why not just add the annotation on the cluster namespace instead of managedcluster?

Reply:
only add the annotation to the ns